openhtf/util/multicast.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Multicast facilities for sending and receiving messages.
+
+This module includes both a MulticastListener that listens on a multicast
+socket and invokes a callback function for each message recieved, and a send()
+function that is used to send one-shot messages to a multicast socket.
+"""
+
+
+import logging
+import socket
+import struct
+import threading
+
+
+_LOG = logging.getLogger(__name__)
+
+DEFAULT_ADDRESS = '239.1.1.1'
+DEFAULT_PORT = 10000
+DEFAULT_TTL = 1
+MAX_MESSAGE_BYTES = 1024  # Maximum allowable message length in bytes.
+PING_STRING = 'OPENHTF_PING'
```

Comment:
this pair of strings is never actually used in this module.  in fact, they're really more coupled to the callback that gets passed on, so should probably not be in this module.

nice refactor pulling this into a separate module